src/Symfony/Component/Form/Extension/Core/Type/CurrencyType.php

```diff
@@ -32,14 +32,39 @@ public function configureOptions(OptionsResolver $resolver): void
 
                 $choiceTranslationLocale = $options['choice_translation_locale'];
 
+                // When the `active` and `legal_tender` options are null, it is unnecessary no need to retrieve the currencies from the meta indice.
+                if (null !== ($active = $options['active']) && null !== ($legalTender = $options['legal_tender'])) {
+                    $filteredCurrencyNames = [];
+
+                    foreach (Currencies::getCurrencyCodes() as $code) {
+                        if (!Currencies::isValidInAnyCountry($code, $legalTender, $active, $options['date'])) {
+                            continue;
+                        }
+
+                        $filteredCurrencyNames[$code] = Currencies::getName($code, $choiceTranslationLocale);
+                    }
+
+                    return ChoiceList::loader(
+                        $this,
+                        new IntlCallbackChoiceLoader(static fn () => array_flip($filteredCurrencyNames)),
+                        $choiceTranslationLocale
+                    );
+                }
+
                 return ChoiceList::loader($this, new IntlCallbackChoiceLoader(static fn () => array_flip(Currencies::getNames($choiceTranslationLocale))), $choiceTranslationLocale);
             },
             'choice_translation_domain' => false,
             'choice_translation_locale' => null,
+            'active' => null,
+            'legal_tender' => null,
+            'date' => new \DateTimeImmutable('today', new \DateTimeZone('Etc/UTC')),
             'invalid_message' => 'Please select a valid currency.',
         ]);
 
         $resolver->setAllowedTypes('choice_translation_locale', ['null', 'string']);
+        $resolver->setAllowedTypes('active', ['bool', 'null']);
```

Comment:
what about removing the "date" option this way?
also, I don't think a boolean makes sense:
```suggestion
        $resolver->setAllowedTypes('valid_at', ['null', \DateTimeInterface::class]);
```

Reply:
should we support ranges of dates BTW?
valid_from and valid_until?

Reply:
~~Regarding the parameters, I think I can change the `active` option to `valid_at` to avoid having 2 parameters. I'll see what I can do. 👍🏻~~ see https://github.com/symfony/symfony/pull/61837#discussion_r2378825857

For the ranges of dates we could, but I haven't included that in the filtering methods. For me, it's already quite good that we no longer have expired currencies. Do you have an use case in mind where a ranges of dates could be useful ?

Reply:
I don't, but I thought it might be easy to add

Reply:
It could be useful for an use case such as :

“Which currencies were legal tender at any point during Q2 2021?”

But the additional complexity would be to handle if the given dates are only overlapping or if the dates have full coverage. This would add another flag (for instance `mode` that could take either `overlap` or `cover`.

Reply:
OK, let's keep it one date :)

Reply:
👍🏻